app/src/workflow.js

```diff
@@ -46,11 +47,27 @@ workflow = function (selection) {
                 "$" +
                 d.outputIndex;
         }
-        conn.selectAll("path").data(workflow.connections, connectionKey).enter().append("path")
+        path = conn.selectAll("path").data(workflow.connections, connectionKey).enter().append("path")
             .style("stroke", strokeColor)
             .style("stroke-width", 10)
             .style("fill", "none")
             .attr("d", connectionPath);
+
+        if (write) {
+            path.on("mouseover", function (d) {
+                d3.select(this).style("stroke", "crimson");
+            });
+            path.on("mouseout", function (d) {
+                d3.select(this).style("stroke", strokeColor);
+            });
+            path.on("click", function (d) {
+                var r = confirm("Remove this connection from this workflow?");
+                if (r === true) {
+                    d3.select(this).remove();
```

Comment:
Something else needs to happen here to delete the connection from the "model", this only deletes it from the "view". You need to find the connection in the workflow.connections list and remove it. You can see the bug if you delete a connection then save the workflow - the connections will reappear when the view is reset.